modules/apps/captcha/captcha-api/src/main/java/com/liferay/captcha/hcaptcha/HCaptchaImpl.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.captcha.hcaptcha;
+
+import com.liferay.captcha.simplecaptcha.SimpleCaptchaImpl;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.captcha.Captcha;
+import com.liferay.portal.kernel.captcha.CaptchaConfigurationException;
+import com.liferay.portal.kernel.captcha.CaptchaException;
+import com.liferay.portal.kernel.exception.SystemException;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONException;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.portal.kernel.util.HttpUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.io.IOException;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Tagnaouti Boubker
+ * @author Jorge Ferrer
+ * @author Brian Wing Shun Chan
+ * @author Daniel Sanz
+ * @author Manuele Castro
+ */
+@Component(
+	configurationPid = "com.liferay.captcha.configuration.CaptchaConfiguration",
+	property = "captcha.engine.impl=com.liferay.captcha.hcaptcha.HCaptchaImpl",
+	service = Captcha.class
+)
+public class HCaptchaImpl extends SimpleCaptchaImpl {
```

Comment:
in this case, SimpleCaptchaImpl (which implements the Captcha interface) is being extended.